pkg/cmd/cache/shared/shared.go

```diff
@@ -71,3 +83,19 @@ pagination:
 
 	return result, nil
 }
+
+func (c *Cache) ExportData(fields []string) map[string]interface{} {
+	v := reflect.ValueOf(c).Elem()
+	fieldByName := func(v reflect.Value, field string) reflect.Value {
+		return v.FieldByNameFunc(func(s string) bool {
+			return strings.EqualFold(field, s)
+		})
+	}
+	data := map[string]interface{}{}
+
+	for _, f := range fields {
+		data[f] = fieldByName(v, f).Interface()
+	}
+
+	return data
+}
```

Comment:
@siketyan : what tests might you add to [`/pkg/cmd/cache/shared/shared_test.go`](https://github.com/cli/cli/blob/c5a7bf8f45440488cc596b21ef396c6d02fa6692/pkg/cmd/cache/shared/shared_test.go) similar to the ones in [`/api/export_pr_test.go`](https://github.com/cli/cli/blob/c5a7bf8f45440488cc596b21ef396c6d02fa6692/api/export_pr_test.go) to test these changes out?

Reply:
@andyfeller Thank you for your review. Added in 07fb228.